rpc/client.go

```diff
@@ -101,11 +103,20 @@ type response[T serde.ResponsePayload] struct {
 // NewRpcClient creates a new RpcClient
 func NewRpcClient(trans transport.Requester) (RpcClientApi, error) {
 	ctx, cancel := context.WithCancel(context.Background())
-
-	c := &rpcClient{trans, &safesync.Map[chan struct{}]{}, &safesync.Map[chan query.LedgerChannelInfo]{}, &safesync.Map[chan query.PaymentChannelInfo]{}, cancel, &sync.WaitGroup{}, common.Address{}, slog.Default()}
+	c := &rpcClient{
+		transport:             trans,
+		completedObjectives:   &safesync.Map[chan struct{}]{},
+		ledgerChannelUpdates:  &safesync.Map[chan query.LedgerChannelInfo]{},
+		paymentChannelUpdates: &safesync.Map[chan query.PaymentChannelInfo]{},
+		cancel:                cancel,
+		wg:                    &sync.WaitGroup{},
+		nodeAddress:           common.Address{},
+		authTokenReady:        &sync.WaitGroup{},
+		logger:                slog.Default(),
+	}
 
 	// Retrieve the address and set it on the rpcClient
-	res, err := waitForRequest[serde.NoPayloadRequest, common.Address](c, serde.GetAddressMethod, serde.NoPayloadRequest{})
+	res, err := authenticatedWaitForRequest[serde.NoPayloadRequest, common.Address](c, serde.GetAddressMethod, serde.NoPayloadRequest{})
```

Comment:
Nit: slightly confusing to call an authorized request method here before getting the auth token. Inside, it will wait on the `c.authToken` wait group which has not yet been "primed" by adding to it. Could/should we replace this call with an `unauthorizedWaitForRequest` to cut out that step?

Reply:
This is a good point. Switched to unauthenticated request in 7df7f7ea.